### PR TITLE
skeleton for periodic cleanup task

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: java/arcs/sdk/android/storage/service/StorageService.kt [hunk 1]
```diff
@@ -46,6 +51,15 @@ class StorageService : ResurrectorService() {
     override fun onCreate() {
         super.onCreate()
         startTime = startTime ?: System.currentTimeMillis()
+
+        val periodicCleanupTask =
+                PeriodicWorkRequest.Builder(PeriodicCleanupTask::class.java, 1, TimeUnit.HOURS)
+                        .addTag(PeriodicCleanupTask.WORKER_TAG)
+                        .build()
```
Comment (jasonwyatt) [inline java/arcs/sdk/android/storage/service/StorageService.kt]: Continuation indentation in kotlin is only 4 spaces (one tab), not two-tabs like in java.

File: java/arcs/sdk/android/storage/service/StorageService.kt [hunk 2]
```diff
@@ -46,6 +51,15 @@ class StorageService : ResurrectorService() {
...
+
+        WorkManager.getInstance(this).enqueueUniquePeriodicWork(
+                PeriodicCleanupTask.WORKER_TAG, ExistingPeriodicWorkPolicy.KEEP, periodicCleanupTask
```
Comment (jasonwyatt) [inline java/arcs/sdk/android/storage/service/StorageService.kt]: reduce indent by 4 spaces, then I recommend using alt+enter: "place arguments on separate lines"

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package arcs.android.storage.ttl
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import arcs.core.util.TaggedLog
+
+class PeriodicCleanupTask(
```
Comment (jasonwyatt) [inline java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]: KDoc please :)

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    override fun doWork(): Result {
+        log.debug { "Running." }
+        // TODO (b/146012246): find and delete expired entities.
+
+        // Indicate whether the task finished successfully with the Result
+        return Result.success()
+    }
+
+    companion object {
+        const val WORKER_TAG = "PeriodicCleanupTask"
```
Comment (jasonwyatt) [inline java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]: KDoc would be good here too.

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+class PeriodicCleanupTask(
+    appContext: Context,
+    workerParams: WorkerParameters
+) : Worker(appContext, workerParams) {
+
+    private val log = TaggedLog { this.toString() }
+    init {
+        log.debug { "Created." }
+    }
```
Comment (jasonwyatt) [inline java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]:
```kotlin
    private val log = TaggedLog { "PeriodCleanupTask" }
    
    init { log.debug { "Created." } }
```

File: java/arcs/sdk/android/storage/service/StorageService.kt (hunk 1 above)
Comment (mmandlis) [reply java/arcs/sdk/android/storage/service/StorageService.kt]: done

File: java/arcs/sdk/android/storage/service/StorageService.kt (hunk 2 above)
Comment (mmandlis) [reply java/arcs/sdk/android/storage/service/StorageService.kt]: done

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt (hunk 3 above)
Comment (mmandlis) [reply java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]: done, sorry :(

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt (hunk 4 above)
Comment (mmandlis) [reply java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]: done

File: java/arcs/android/storage/ttl/PeriodicCleanupTask.kt (hunk 5 above)
Comment (mmandlis) [reply java/arcs/android/storage/ttl/PeriodicCleanupTask.kt]: a sad story how copy&paste could fail you, if you copy from the wrong place. fixed.

**Pull request merged**